numba/cgutils.py

```diff
@@ -47,6 +48,117 @@ def make_bytearray(buf):
     n = len(b)
     return ir.Constant(ir.ArrayType(ir.IntType(8), n), b)
 
+_struct_proxy_cache = {}
+
+
+def create_struct_proxy(fe_type):
+    """
+    Returns a specialized StructProxy subclass for the given fe_type.
+    """
+    res = _struct_proxy_cache.get(fe_type)
+    if res is None:
+        clsname = StructProxy.__name__ + '_' + str(fe_type)
+        bases = (StructProxy,)
+        clsmembers = dict(_fe_type=fe_type)
+        res = type(clsname, bases, clsmembers)
+        _struct_proxy_cache[fe_type] = res
+    return res
+
+
+class StructProxy(object):
+    """
+    Creates a `Structure` like interface that is constructed with information
+    from DataModel instance.  FE type must have a data model that is a
```

Comment:
i used FE to imply any high level "type" instance with no(or little) information about layout.  In the scope of Numba, FE type == Numba type.  (but maybe the whole package can be reused in some other experimental language project ;-) )